src/otx/core/data/transform_libs/mmaction.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Helper to support MMAction data transform functions."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable
+
+from mmaction.datasets.transforms import PackActionInputs as MMPackActionInputs
+from mmaction.datasets.transforms import SampleFrames as MMSampleFrames
+from mmaction.registry import TRANSFORMS
+from torchvision import tv_tensors
+
+from otx.core.data.entity.action import ActionClsDataEntity
+from otx.core.data.entity.base import ImageInfo
+from otx.core.utils.config import convert_conf_to_mmconfig_dict
+
+if TYPE_CHECKING:
+    import numpy as np
+    from mmengine.registry import Registry
+
+    from otx.core.config.data import SubsetConfig
+
+
+@TRANSFORMS.register_module(force=True)
+class LoadVideo:
+    """Class to convert OTXDataEntity to dict for MMAction framework."""
+
+    def __call__(self, entity: ActionClsDataEntity) -> dict:
+        """Transform ActionClsDataEntity to MMAction data dictionary format."""
+        video: list[np.ndarray] = entity.image
+
+        results: dict[str, Any] = {}
+        results["start_index"] = 0
+        results["total_frames"] = len(video)
+        results["modality"] = "RGB"
+        results["imgs"] = video
+        results["img_shape"] = video[0].shape[:2]
+        results["ori_shape"] = video[0].shape[:2]
+        results["__otx__"] = entity
+
+        return results
+
+
+@TRANSFORMS.register_module(force=True)
+class SampleFrames(MMSampleFrames):
+    """Class to override SampleFrames.
+
+    MMAction's SampleFrames just sample frame indices for training.
+    Actual frame sampling is done by decode pipeline.
+    However, OTX already has decoded data, so here, actual sampling frame will be conducted.
```

Comment:
I think that the current implementation to grab frames from an video file has a problem since it is memory unbounded.

To make it memory bounded, you should move the frame grabbing after sampling frame indices. You might be able to grab the frames from the sampled indices with `ActionClsDataEntity.video` (Please see my another comment in `core/data/entity/action.py`).

Reply:
Done by 93aed91